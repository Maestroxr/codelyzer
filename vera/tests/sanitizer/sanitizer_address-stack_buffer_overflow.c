#include <stdlib.h>
#include <stdio.h>

int main(int argc, char** argv) 
{
	int res = -1;
	int stackArray[ 100 ] = { 0 };
	if (stackArray[ argc+ 100 ])
	{
		// BOOM Stack-buffer-overflow
		res = stackArray[ argc + 100 ];  // BOOM Stack-buffer-overflow
	}
	return res;
}