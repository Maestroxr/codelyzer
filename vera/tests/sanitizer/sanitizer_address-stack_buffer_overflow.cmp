C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-stack_buffer_overflow.c:12: Memory:[C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-stack_buffer_overflow]: stack-buffer-overflow, READ, variable identifier: stackArray.
