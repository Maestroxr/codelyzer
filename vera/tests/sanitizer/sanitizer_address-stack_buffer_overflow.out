C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-stack_buffer_overflow.c:12: Memory:[C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-stack_buffer_overflow]: Stack buffer overflow, suspect variables: ['stackArray']
	Was accessed by read of size 4 at:
		#0 sanitizer_address-stack_buffer_overflow.c:12 in function 'main'
	Accessed address is located in stack of thread T0 at offset 692 in frame:
		#0 sanitizer_address-stack_buffer_overflow.c:5 in function 'main'
	This frame has 5 object(s):
    [32, 36) ''
    [96, 100) ''
    [160, 168) ''
    [224, 228) 'res'
    [288, 688) 'stackArray' <== Memory access at offset 692 overflows this variable

